rmgpy/solver/surface.pyx

```diff
@@ -384,17 +402,31 @@ cdef class SurfaceReactor(ReactionSystem):
 
         C = np.zeros_like(self.core_species_concentrations)
         V = self.V  # constant volume reactor
+        A = self.V * surface_volume_ratio_si  # area
+        total_sites = self.surface_site_density.value_si * A  # todo: double check units
 
+        current_surface_coverages = {}
         for j in range(num_core_species):
             if species_on_surface[j]:
                 C[j] = (N[j] / V) / surface_volume_ratio_si
+                surface_site_fraction = N[j] / total_sites
+                species = cov_dep_index_species[j]
+                current_surface_coverages[species] = surface_site_fraction
             else:
                 C[j] = N[j] / V
             #: surface species are in mol/m2, gas phase are in mol/m3
             core_species_concentrations[j] = C[j]
 
         for j in range(ir.shape[0]):
             k = kf[j]
+            if j in cov_dep:
+                cov_dep_j = cov_dep[j]
+                for species, parameters in cov_dep_j.items():
+                    if species in current_surface_coverages:
```

Comment:
I think this should be fine if we use the `make_new_species` function for each species in the cov dep dict.  Drawback there is that you would be adding species to your model that may not have been added through reactions.  If we decide we don't want to do it this way, we will have to implement some kind of mapping here